File: app/src/main/java/com/example/migym/workers/WorkoutNotificationWorker.java
package com.example.migym.workers;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import androidx.annotation.NonNull;
import androidx.core.app.NotificationCompat;
import androidx.work.Worker;
import androidx.work.WorkerParameters;
import com.example.migym.MainActivity;
import com.example.migym.R;

public class WorkoutNotificationWorker extends Worker {
    private static final String CHANNEL_ID = "workout_notifications";
    private static final int NOTIFICATION_ID = 1;

    public WorkoutNotificationWorker(@NonNull Context context, @NonNull WorkerParameters params) {
        super(context, params);
    }

    @Override
    public Result doWork() {
        String workoutTitle = getInputData().getString("workout_title");
        String notificationType = getInputData().getString("notification_type");

        if (workoutTitle == null || notificationType == null) {
            return Result.failure();
        }

        createNotificationChannel();

        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        PendingIntent pendingIntent = PendingIntent.getActivity(
                getApplicationContext(),
                0,
                intent,
                PendingIntent.FLAG_IMMUTABLE
        );

        String title;
        String content;
        if (notificationType.equals("reminder")) {
            title = getApplicationContext().getString(R.string.workout_reminder);
            content = String.format(getApplicationContext().getString(R.string.workout_reminder_content), workoutTitle);
        } else {
            title = getApplicationContext().getString(R.string.workout_completed);
            content = String.format(getApplicationContext().getString(R.string.workout_completed_content), workoutTitle);
        }

        NotificationCompat.Builder builder = new NotificationCompat.Builder(getApplicationContext(), CHANNEL_ID)
                .setSmallIcon(R.drawable.ic_notification)
                .setContentTitle(title)
                .setContentText(content)
                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                .setAutoCancel(true)
                .setContentIntent(pendingIntent);

        NotificationManager notificationManager = (NotificationManager) getApplicationContext()
                .getSystemService(Context.NOTIFICATION_SERVICE);

        notificationManager.notify(NOTIFICATION_ID, builder.build());

        return Result.success();
    }

    private void createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            CharSequence name = getApplicationContext().getString(R.string.channel_name);
            String description = getApplicationContext().getString(R.string.channel_description);
            int importance = NotificationManager.IMPORTANCE_DEFAULT;
            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, name, importance);
            channel.setDescription(description);

            NotificationManager notificationManager = getApplicationContext()
                    .getSystemService(NotificationManager.class);
            notificationManager.createNotificationChannel(channel);
        }
    }
} 

File: app/src/main/java/com/example/migym/workers/WorkoutNotificationWorker.kt
package com.example.dasproyectonew.workers

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.os.Build
import androidx.core.app.NotificationCompat
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.example.dasproyectonew.R

class WorkoutNotificationWorker(
    private val context: Context,
    params: WorkerParameters
) : Worker(context, params) {

    override fun doWork(): Result {
        val workoutTitle = inputData.getString(KEY_WORKOUT_TITLE) ?: return Result.failure()
        val notificationType = inputData.getString(KEY_NOTIFICATION_TYPE) ?: return Result.failure()

        createNotificationChannel()

        val notification = when (notificationType) {
            TYPE_REMINDER -> createReminderNotification(workoutTitle)
            TYPE_COMPLETION -> createCompletionNotification(workoutTitle)
            else -> return Result.failure()
        }

        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        notificationManager.notify(System.currentTimeMillis().toInt(), notification)

        return Result.success()
    }

    private fun createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(
                CHANNEL_ID,
                "Workout Notifications",
                NotificationManager.IMPORTANCE_HIGH
            ).apply {
                description = "Notifications for workout reminders and completion"
            }
            val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            notificationManager.createNotificationChannel(channel)
        }
    }

    private fun createReminderNotification(workoutTitle: String) = NotificationCompat.Builder(context, CHANNEL_ID)
        .setSmallIcon(R.drawable.ic_notification)
        .setContentTitle("Upcoming Workout")
        .setContentText("Your $workoutTitle starts in 1 hour!")
        .setPriority(NotificationCompat.PRIORITY_HIGH)
        .setAutoCancel(true)
        .build()

    private fun createCompletionNotification(workoutTitle: String) = NotificationCompat.Builder(context, CHANNEL_ID)
        .setSmallIcon(R.drawable.ic_notification)
        .setContentTitle("Workout Complete")
        .setContentText("How was your $workoutTitle? Rate your performance!")
        .setPriority(NotificationCompat.PRIORITY_HIGH)
        .setAutoCancel(true)
        .build()

    companion object {
        const val CHANNEL_ID = "workout_notifications"
        const val KEY_WORKOUT_TITLE = "workout_title"
        const val KEY_NOTIFICATION_TYPE = "notification_type"
        const val TYPE_REMINDER = "reminder"
        const val TYPE_COMPLETION = "completion"
    }
} 